### PR TITLE
Chore: rename table_model.ts => TableModel.ts; update references

### PR DESCRIPTION
**Pull request by user10x**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #
https://github.com/grafana/grafana/issues/51230

**Special notes for your reviewer**:

```
 .betterer.results                                  | 52 +++++++++++-----------
 public/app/core/{table_model.ts => TableModel.ts}  |  0
 .../{table_model.test.ts => TableModel.test.ts}    |  2 +-
 .../app/features/explore/utils/decorators.test.ts  |  2 +-
 public/app/features/plugins/plugin_loader.ts       |  2 +-
 .../datasource/elasticsearch/elastic_response.ts   |  2 +-
 .../plugins/datasource/influxdb/influx_series.ts   |  2 +-
 .../plugins/datasource/influxdb/response_parser.ts |  2 +-
 .../plugins/panel/table-old/specs/renderer.test.ts |  2 +-
 public/app/plugins/panel/table-old/transformers.ts |  2 +-
 public/app/plugins/panel/table-old/types.ts        |  2 +-
 11 files changed, 35 insertions(+), 35 deletions(-)
 rename public/app/core/{table_model.ts => TableModel.ts} (100%)
 rename public/app/core/specs/{table_model.test.ts => TableModel.test.ts} (99%)
```

### CODE REVIEW

File: public/app/features/plugins/plugin_loader.ts [hunk 1]
```diff
@@ -116,7 +116,7 @@ exposeToPlugin('app/core/utils/ticks', ticks);
 exposeToPlugin('app/core/config', config);
 exposeToPlugin('app/core/time_series', TimeSeries);
 exposeToPlugin('app/core/time_series2', TimeSeries);
-exposeToPlugin('app/core/table_model', TableModel);
```
Comment (ryantxu) [inline public/app/features/plugins/plugin_loader.ts:119]: we should be very careful changing the path external things may use.  We could consider removing it, but let's not change it :)

Comment (user10x) [reply public/app/features/plugins/plugin_loader.ts]: ahh! I see! Although at this point I am not able to run this in the UI but reverted the the change so path exposed to plugin remains same. Good catch and thank you

File: public/app/features/plugins/plugin_loader.ts [hunk 2]
```diff
@@ -18,11 +18,11 @@ import * as grafanaData from '@grafana/data';
 import * as grafanaRuntime from '@grafana/runtime';
 import slateReact from '@grafana/slate-react';
 import * as grafanaUIraw from '@grafana/ui';
+import TableModel from 'app/core/TableModel';
 import config from 'app/core/config';
 import { appEvents, contextSrv } from 'app/core/core';
 import { BackendSrv, getBackendSrv } from 'app/core/services/backend_srv';
 import impressionSrv from 'app/core/services/impression_srv';
-import TableModel from 'app/core/table_model';
```
Comment (ryantxu) [inline public/app/features/plugins/plugin_loader.ts:25]: 👍

**Comment by user10x** [comment]
@ivanahuckova review please :) There's other PRs attached with issue, do not want to spam your notifications

**Comment by ivanahuckova** [comment]
@user10x In https://github.com/grafana/grafana/issues/51230 we have specified folders in which we would like to change the casing. This is not part of it. And it is and this part of code base that other squad maintains, so they need to review this.

**Review by ashharrison90 — APPROVED** [review]
lgtm, thanks for this! 🙌 

i've merged main and pushed it to resolve the betterer conflict - hope you don't mind 😄

**Pull request merged**